tests/specs/verification.k

```diff
@@ -35,6 +37,51 @@ module VERIFICATION
      andBool (B >Int log2Int(I) andBool 64 >Int B)
     [simplification]
 
+////////////////////////////////////////////////////////////////////////////////
+// Recombine 128-bit Ints
+////////////////////////////////////////////////////////////////////////////////
+
+//      Upper 64 bits                    Lower 64 bits
+  rule ((X:Int >>Int 64) <<Int 64) +Int (X &Int 18446744073709551615) => X [simplification]
+
+  rule X:Int >>Int N:Int => 0 
+    requires X <Int 2 ^Int N
+     andBool 0 <=Int X [simplification]
+
+  rule X:Int >Int Y:Int /Int Z:Int => true
+    requires X >Int Y
+     andBool Z >Int 0 [simplification]
+
+  rule X:Int /Int Y:Int >=Int 0 => true
+    requires X >=Int 0
+     andBool Y >=Int 0 [simplification]
+
+  rule 0 >Int X:Int /Int Y:Int => false
+    requires X >=Int 0
+     andBool Y >=Int 0 [simplification]
+
+  rule A:Int >Int (X:Int &Int B:Int) => true 
+    requires X >=Int 0
+     andBool A >=Int 0
+     andBool B >=Int 0
+     andBool A >Int B [simplification]
+
+  rule 0 >Int (X:Int &Int Y:Int) => 0 >Int X
+    requires Y >=Int 0 [simplification]
+  
+  rule X:Int &Int Y:Int >=Int 0 => true
+    requires X >=Int 0
+     andBool Y >=Int 0  [simplification]
+
+  syntax Bool ::= allOnes(Int) [function]
+  rule allOnes(0) => true
+  rule allOnes(N) => allOnes(N /Int 2) requires N modInt 2 ==Int 1 andBool N >Int 0
+  rule allOnes(N) => false requires N modInt 2 ==Int 0 andBool N >Int 0
+
+  rule X:Int &Int Y:Int => X
+    requires allOnes(Y)
+     andBool X <=Int Y [simplification]
+
```

Comment:
Changes included from pact-verification-dev: I need to categorize thees new lemmas and double-check for correctness.